app/src/main/java/com/github/geohunt/app/ui/components/profile/ChallengeGrid.kt

```diff
@@ -32,21 +32,15 @@ fun ChallengeGrid(challenges: List<LazyRef<Challenge>>) {
 }
 
 @Composable
-private fun ChallengeItem(challenge: LazyRef<Challenge>) {
-    val c = rememberLazyRef { challenge }
-
-    if (c.value == null) {
-        Box(modifier = Modifier.fillMaxSize().background(skeleton_loading_background))
-    } else {
-        AsyncImage(
-            model = ImageRequest.Builder(LocalContext.current)
-                .data("https://picsum.photos/430/400") // TODO Integrate with file store
-                .size(Size.ORIGINAL)
-                .crossfade(true)
-                .build(),
-            contentDescription = "Challenge",
-            modifier = Modifier
-                .padding(8.dp)
-        )
-    }
+private fun ChallengeItem(challenge: Challenge) {
+    AsyncImage(
+        model = ImageRequest.Builder(LocalContext.current)
+            .data(challenge.photoUrl) // TODO Integrate with file store
```

Comment:
I believe this is enough to be integrated with file store

Reply:
I mean the whole point is that coil is doing it's magic (F in the chat for my beautiful `LazyRef<Bitmap>`)

Reply:
The TODO stayed but it was about the older implementation of the ChallengeGrid! My bad.